crates/kubelet/src/provider/mod.rs

```diff
@@ -110,6 +110,25 @@ pub trait Provider: Sized + Send + Sync + 'static {
     // TODO: Is there a way to provide a default implementation of this if Self::PodState: Default?
     async fn initialize_pod_state(&self, pod: &Pod) -> anyhow::Result<Self::PodState>;
 
+    /// Hook to allow the provider to react to the Kubelet being shut down
+    ///
+    /// It receives only the node name as a parameter in case the provider wants to set a condition
+    /// on the object - for example to to signify that it did not crash but performed an orderly
+    /// shutdown.
+    ///
+    /// There are currently no mechanisms in place to propagate the shutdown trigger to the state
+    /// machine, providers will have to implement this themselves via a [`std::sync::mpsc::channel`]
+    /// or the shared state or something similar.
+    ///
+    /// # Arguments
+    ///
+    /// * `node_name` - The name of the node object that was created by this Kubelet instance
+    ///
+    async fn shutdown(&self, node_name: &str) -> anyhow::Result<()> {
```

Comment:
So if we wanted to implement something like the new behavior in 1.21, we could send the `stop()` command to our Pod handles, without instructing the Pods to shut down via the Kubernetes API. We probably can't wait for the state machines to exit though...

Reply:
I mean, you technically could wait on the state machines. But it depends on how you have things wired up